code/modules/hydroponics/compost_bin.dm

```diff
@@ -0,0 +1,195 @@
+#define BASE_BIOMASS_CAPACITY 1500
+#define BASE_COMPOST_CAPACITY 1500
+#define SOIL_COST 25
+#define SOIL /obj/item/stack/sheet/soil
+#define DECAY 0.2
+#define MIN_CONVERSION 10
+/**
+  * # compost bin
+  * used to make soil from plants.
+  * Doesn't have components.
+  */
+/obj/machinery/compost_bin
```

Comment:
Since this doesnt use power at all, it should probably be `/obj/structure/compost_bin` instead.

Reply:
I commented on another convo, but to reiterate, I made this to act as a no-power biogen replacement that can (for now) only make soil. Aside from the lack of power requirement, I think it behaves more like a machine, so I'd rather keep it as such.